webapp/src/Service/DOMJudgeService.php

```diff
@@ -320,6 +320,53 @@ public function getUnreadClarifications(): array
         return $unreadClarifications;
     }
 
+    /**
+     * @return array<array{submissionid: int, judgingid: int, body: string}>
+     */
+
+    public function getUnreadJudgements(): array
+    {
+        $user = $this->getUser();
+        $team = $user->getTeam();
+        $contest = $this->getCurrentContest($team->getTeamId());
+        $unreadJudgements = [];
+        if ($contest === null) {
+            return $unreadJudgements;
+        }
+
+        $queryBuilder = $this->em->createQueryBuilder()
+            ->from(Judging::class, 'j')
+            ->select('j, c, s')
+            ->leftJoin('j.contest', 'c')
+            ->leftJoin('j.submission', 's')
+            ->groupBy('j.judgingid')
+            ->orderBy('j.judgingid')
+            ->andWhere('j.contest = :cid')
+            ->setParameter('cid', $contest->getCid())
+            ->andWhere('j.result IS NOT NULL')
+            ->andWhere('s.team = :team')
+            ->setParameter('team', $team)
+            ->andWhere('s.submittime < c.endtime')
+            ->andWhere('j.valid = 1')
+            ->andWhere('j.seen = 0');
```

Comment:
is there a migration missing? `seen` doesn't exist yet, does it?

Reply:
BTW, this SQL query is the same as the one a user would use to access `/api/contests/:cid/judgements`, except it filters out entries they have never seen. 

Therefore, it is also feasible to implement it directly on the frontend js instead of adding a function to the `/update` endpoint.

Reply:
> When are we setting the `seen` to true? I assume we already do this.

https://github.com/DOMjudge/domjudge/blob/4a991717abdabaad68aa203135b04d9110f7312a/webapp/templates/team/partials/submission_list.html.twig#L33-L49

When click submission and see detail in team home page.